test/integration/src/test/java/io/pravega/test/integration/BackUpRecoveryTest.java

```diff
@@ -0,0 +1,857 @@
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.watermarks.Watermark;
+import io.pravega.storage.filesystem.FileSystemSimpleStorageFactory;
+import io.pravega.storage.filesystem.FileSystemStorageConfig;
+import io.pravega.test.common.AssertExtensions;
+import io.pravega.test.common.ThreadPooledTestSuite;
+import io.pravega.test.integration.utils.LocalServiceStarter;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static io.pravega.common.concurrent.Futures.loop;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Integration test to verify data recovery.
+ * Recovery scenario: when data written to Pravega is already flushed to the long term storage.
+ * Tests replicate different environments for data recovery.
+ */
+@Slf4j
+public class BackUpRecoveryTest extends ThreadPooledTestSuite {
+    private static final Duration TIMEOUT = Duration.ofMillis(100 * 10000);
+    private static final Duration READ_TIMEOUT = Duration.ofMillis(1000);
+    private static final Duration TRANSACTION_TIMEOUT = Duration.ofMillis(10000);
+
+    /**
+     * Write {@link #TOTAL_NUM_EVENTS} events to verify recovery.
+     */
+    private static final int TOTAL_NUM_EVENTS = 300;
+
+    private static final Random RANDOM = new Random(1234);
+
+    /**
+     * Scope and streams to read and write events.
+     */
+    private static final String SCOPE = "testRecoveryScope";
+    private static final String STREAM1 = "testRecoveryStream" + RANDOM.nextInt(Integer.MAX_VALUE);
+    private static final String EVENT = "12345";
+    private final ScalingPolicy scalingPolicy = ScalingPolicy.fixed(1);
+    private final StreamConfiguration config = StreamConfiguration.builder().scalingPolicy(scalingPolicy).build();
+
+    private final AtomicLong timer = new AtomicLong();
+
+    private FileSystemSimpleStorageFactory storageFactory;
+
+    private BookKeeperLogFactory dataLogFactory;
```

Comment:
This appears to be initialized by each test individually. It can be made into a local.